### PR TITLE
Add document about releases for GA and post GA [SKIP CI].

### LINKED ISSUE #1797: Finalize the nomenclature for stable and edge releases
Finalize and document the nomenclature for stable and edge releases

### PR DESCRIPTION
**Pull request by lipingxue**
Fixed #1797

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -277,6 +277,24 @@ COVERAGE_PROCESS_START DEFAULT=/coverage.rc
...
 
+## Releases for GA and post GA
+The following is the plan for releases for GA and post GA. Before GA, the release name format remains the same.
+
+### Type of Release:
+We plan to have two types of releases: edge release and stable release.
+
+edge release: will be released every month for users who want a drop of latest.
+
+stable release: will be release every 3 month for users who want to a stable version.
```
Comment (shaominchen) [inline CONTRIBUTING.md]: want to => want

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -277,6 +277,24 @@ COVERAGE_PROCESS_START DEFAULT=/coverage.rc
...
+
+### Nomenclature for release:
+
+The Nomenclature for release will follow the format ```major-version. minor-version.build-version.release-type```.
```
Comment (shaominchen) [inline CONTRIBUTING.md]: Remove the redundant space after "major-version."

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -277,6 +277,24 @@ COVERAGE_PROCESS_START DEFAULT=/coverage.rc
...
+
+“Major-Version” will be the major version number, “minor-version” will be the minor version number which will be bumped on each monthly release. “build-version” by default is 0. If we decide to cut one more release on that month (maybe some patch releases), we can increase the “build-version” by 1 for that release. “release-type” has two possible values: “stable” to mark a stable release and “edge” to mark a edge release.
```
Comment (shaominchen) [inline CONTRIBUTING.md]: 1. will be bumped on each monthly release => will be bumped up for each monthly release
2. on that month => in that month
3. a edge => an edge

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -277,6 +277,24 @@ COVERAGE_PROCESS_START DEFAULT=/coverage.rc
...
+
+For Example, assume we plan to cut the first release for GA on August , the release number will be “1.0.0.stable”. Then the monthly edge release on September will be “1.1.0.edge”, the monthly release on October will be “1.2.0.edge” and the stable release on November will be “1.3.0.stable”, etc.
```
Comment (shaominchen) [inline CONTRIBUTING.md]: 1. on August/September/October/November => in August/...
2. Remove the redundant space after "August"

File: docs/user-guide/faq.md [hunk 5]
```diff
@@ -19,6 +19,15 @@ This can be achieved via [Tenancy](http://vmware.github.io/docker-volume-vsphere
 ### Can I migrate data between Linux and Windows containers?
 Volumes created via the Linux plugin are formatted with ext4 by default, and the ones created via the Windows plugin are formatted with NTFS. While it is possible to cross-mount such volumes, the vSphere Storage for Docker plugin doesn't support such cases, nor does it provide any explicit help.
 
+### Which release I should use?
+You can choose release based on your needs. There will be two types of releases: edge release and stable release.
+
+edge release: will be released every month for users who want a drop of latest.
+
+stable release: will be release every 3 month for users who want to a stable version.
```
Comment (shaominchen) [inline docs/user-guide/faq.md]: want to => want

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -277,6 +277,24 @@ COVERAGE_PROCESS_START DEFAULT=/coverage.rc
 
 The CI/CD system is based on [Drone platform](https://drone.io/) and the server is  https://ci.vmware.run/. More information is found at our [CI.md](https://github.com/vmware/docker-volume-vsphere/blob/master/CI.md)
 
+## Releases for GA and post GA
```
Comment (shuklanirdesh82) [inline CONTRIBUTING.md]: nit: "vDVS release naming convention" or "Release naming convention"

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -277,6 +277,24 @@ COVERAGE_PROCESS_START DEFAULT=/coverage.rc
...
+The following is the plan for releases for GA and post GA. Before GA, the release name format remains the same.
```
Comment (shuklanirdesh82) [inline CONTRIBUTING.md]: Not sure we should call out GA/Post GA, there will be a question from customer side about GA date. 

/CC @tusharnt

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -277,6 +277,24 @@ COVERAGE_PROCESS_START DEFAULT=/coverage.rc
...
+
+### Type of Release:
+We plan to have two types of releases: edge release and stable release.
+
+edge release: will be released every month for users who want a drop of latest.
```
Comment (shuklanirdesh82) [inline CONTRIBUTING.md]: edge => Edge

File: CONTRIBUTING.md (hunk 1 above)
Comment (shuklanirdesh82) [inline CONTRIBUTING.md]: stable => Stable

File: CONTRIBUTING.md (hunk 2 above)
Comment (shuklanirdesh82) [inline CONTRIBUTING.md]: The Nomenclature => The nomenclature

File: docs/user-guide/faq.md [hunk 9]
```diff
@@ -19,6 +19,15 @@ This can be achieved via [Tenancy](http://vmware.github.io/docker-volume-vsphere
 ### Can I migrate data between Linux and Windows containers?
 Volumes created via the Linux plugin are formatted with ext4 by default, and the ones created via the Windows plugin are formatted with NTFS. While it is possible to cross-mount such volumes, the vSphere Storage for Docker plugin doesn't support such cases, nor does it provide any explicit help.
 
+### Which release I should use?
+You can choose release based on your needs. There will be two types of releases: edge release and stable release.
+
+edge release: will be released every month for users who want a drop of latest.
```
Comment (shuklanirdesh82) [inline docs/user-guide/faq.md]: same as above

File: CONTRIBUTING.md (hunk 1 above)
Comment (ashahi1) [inline CONTRIBUTING.md]: will be release >> will be released ?

File: docs/user-guide/faq.md [hunk 10]
```diff
@@ -19,6 +19,15 @@ This can be achieved via [Tenancy](http://vmware.github.io/docker-volume-vsphere
 ### Can I migrate data between Linux and Windows containers?
 Volumes created via the Linux plugin are formatted with ext4 by default, and the ones created via the Windows plugin are formatted with NTFS. While it is possible to cross-mount such volumes, the vSphere Storage for Docker plugin doesn't support such cases, nor does it provide any explicit help.
 
+### Which release I should use?
+You can choose release based on your needs. There will be two types of releases: edge release and stable release.
```
Comment (shuklanirdesh82) [inline docs/user-guide/faq.md]: how about this?

Edge release: A monthly release with new features and bug fixes
Stable release: A quarterly release with reliable/stable updates

File: docs/user-guide/faq.md [hunk 11]
```diff
@@ -19,6 +19,15 @@ This can be achieved via [Tenancy](http://vmware.github.io/docker-volume-vsphere
...
+
+edge release: will be released every month for users who want a drop of latest.
+
+stable release: will be release every 3 month for users who want to a stable version.
+
+For details about the nomenclature for release, please refer [here](/CONTRIBUTING.md/#releases-for-ga-and-post-ga).
```
Comment (shuklanirdesh82) [inline docs/user-guide/faq.md]: How about this?

Please refer [release convention](https://github.com/lipingxue/docker-volume-vsphere/blob/9d6249d1503f263bc27e16db8c32ee588899aebf/CONTRIBUTING.md#releases-for-ga-and-post-ga) for more details. 

need to supply fully qualified link, I don't think public facing document will resolve `/CONTRIBUTING.md/#re`path.

File: CONTRIBUTING.md (hunk 3 above)
Comment (shuklanirdesh82) [inline CONTRIBUTING.md]: `Major-Version” will be the major version number` :)

can you please elaborate more about major version?

File: CONTRIBUTING.md (hunk 1 above)
Comment (lipingxue) [reply CONTRIBUTING.md]: Done.

File: CONTRIBUTING.md (hunk 2 above)
Comment (lipingxue) [reply CONTRIBUTING.md]: Done

File: CONTRIBUTING.md (hunk 3 above)
Comment (lipingxue) [reply CONTRIBUTING.md]: Done.

File: CONTRIBUTING.md (hunk 4 above)
Comment (lipingxue) [reply CONTRIBUTING.md]: Done.

File: docs/user-guide/faq.md (hunk 5 above)
Comment (lipingxue) [reply docs/user-guide/faq.md]: Done.

File: CONTRIBUTING.md (hunk 6 above)
Comment (lipingxue) [reply CONTRIBUTING.md]: Done.

File: CONTRIBUTING.md (hunk 8 above)
Comment (lipingxue) [reply CONTRIBUTING.md]: Done.

File: CONTRIBUTING.md (hunk 1 above)
Comment (lipingxue) [reply CONTRIBUTING.md]: Done.

File: CONTRIBUTING.md (hunk 2 above)
Comment (lipingxue) [reply CONTRIBUTING.md]: Done.

File: docs/user-guide/faq.md (hunk 9 above)
Comment (lipingxue) [reply docs/user-guide/faq.md]: Done.

File: CONTRIBUTING.md (hunk 1 above)
Comment (lipingxue) [reply CONTRIBUTING.md]: Done.

File: docs/user-guide/faq.md (hunk 10 above)
Comment (lipingxue) [reply docs/user-guide/faq.md]: Done.

File: docs/user-guide/faq.md (hunk 11 above)
Comment (lipingxue) [reply docs/user-guide/faq.md]: Done

File: CONTRIBUTING.md (hunk 3 above)
Comment (lipingxue) [reply CONTRIBUTING.md]: Done

**Comment by lipingxue** [comment]
@shaominchen @shuklanirdesh82 @ashahi1 I have addressed your comments.

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -277,6 +277,24 @@ COVERAGE_PROCESS_START DEFAULT=/coverage.rc
 
 The CI/CD system is based on [Drone platform](https://drone.io/) and the server is  https://ci.vmware.run/. More information is found at our [CI.md](https://github.com/vmware/docker-volume-vsphere/blob/master/CI.md)
 
+## Release naming convention
+The following is the plan for releases for GA and post GA. Before GA, the release name format remains the same.
```
Comment (shuklanirdesh82) [inline CONTRIBUTING.md]: Need to reword as below:

`The following is the plan for GA and post GA releases. Before GA, the release name format remains the same e.g. 0.XX (existing release tag)`

File: CONTRIBUTING.md [hunk 13]
```diff
@@ -277,6 +277,24 @@ COVERAGE_PROCESS_START DEFAULT=/coverage.rc
...
+
+### Type of Release:
+We plan to have two types of releases: edge release and stable release.
+
+Edge release: A monthly release with new features and bug fixes.
+
```
Comment (shuklanirdesh82) [inline CONTRIBUTING.md]: nit: no need for extra line

File: docs/user-guide/faq.md [hunk 14]
```diff
@@ -19,6 +19,15 @@ This can be achieved via [Tenancy](http://vmware.github.io/docker-volume-vsphere
 ### Can I migrate data between Linux and Windows containers?
 Volumes created via the Linux plugin are formatted with ext4 by default, and the ones created via the Windows plugin are formatted with NTFS. While it is possible to cross-mount such volumes, the vSphere Storage for Docker plugin doesn't support such cases, nor does it provide any explicit help.
 
+### Which release I should use?
+You can choose release based on your needs. There will be two types of releases: edge release and stable release.
+
+Edge release: A monthly release with new features and bug fixes.
+
```
Comment (shuklanirdesh82) [inline docs/user-guide/faq.md]: nit: same as above

File: docs/user-guide/faq.md [hunk 15]
```diff
@@ -19,6 +19,15 @@ This can be achieved via [Tenancy](http://vmware.github.io/docker-volume-vsphere
...
+Stable release: A quarterly release with reliable/stable updates.
+
+For details about the nomenclature for release, please refer [here](https://github.com/vmware/docker-volume-vsphere/blob/master/CONTRIBUTING.md#release-naming-convention).
```
Comment (shuklanirdesh82) [inline docs/user-guide/faq.md]: Please reword this as shown below:

```Please refer [release convention](https://github.com/vmware/docker-volume-vsphere/blob/master/CONTRIBUTING.md#release-naming-convention) for more details. ```

View: Please refer [release convention](https://github.com/vmware/docker-volume-vsphere/blob/master/CONTRIBUTING.md#release-naming-convention) for more details.

**Comment by lipingxue** [comment]
@shuklanirdesh82 Could you review it?

**Comment by shuklanirdesh82** [comment]
> @shuklanirdesh82 Could you review it?

@lipingxue I am waiting for the comments to be addressed.

File: CONTRIBUTING.md (hunk 12 above)
Comment (lipingxue) [reply CONTRIBUTING.md]: Done.

File: CONTRIBUTING.md (hunk 13 above)
Comment (lipingxue) [reply CONTRIBUTING.md]: That line is needed, otherwise, the line "edge release" and "stable release" will be in the same line when browsing the md file.

File: docs/user-guide/faq.md (hunk 14 above)
Comment (lipingxue) [reply docs/user-guide/faq.md]: same.

File: docs/user-guide/faq.md (hunk 15 above)
Comment (lipingxue) [reply docs/user-guide/faq.md]: Done.

File: CONTRIBUTING.md (hunk 13 above)
Comment (shuklanirdesh82) [reply CONTRIBUTING.md]: got it. Thanks for checking!
